### PR TITLE
Fix python and dwarf unwind for vllm profiling

### PR DESCRIPTION
**Pull request by kylewanginchina**
### This PR is for:

- Agent

#### Steps to reproduce the bug
- Deploy DeepFlow Agent to perform continuous profiling on vLLM (Python 3.10+ with PyTorch) inference service
- Observe the following issues:
  - DWARF stack unwinding produces function addresses that don't match actual functions, especially in large shared libraries like libcuda.so
  - When the instruction pointer falls within special memory regions like [vdso] or [uprobes], stack unwinding produces incorrect results or crashes
  - Many flame graph frames show [lost] incomplete python c stack, especially in PyTorch nn.Module.__call__ call chains
  - Some Python function frame symbols are missing or garbled
#### Changes to fix the bug
- Correctly calculate ASLR bias offset:
  - Parse the file offset field when reading /proc/<pid>/maps
  - Add calculate_bias() function to correctly compute bias = process_vaddr - elf_vaddr by parsing ELF PT_LOAD program headers
  - Fix edge case bug where the last executable memory area was not added to the result list
- Expand DWARF shard capacity:
  - Increase UNWIND_SHARDS_PER_PROCESS from 256 to 1024 to support large processes (e.g., vLLM loading many .so files)
  - Change process_shard_list_t.len type from uint8_t to uint16_t
  - Add synthetic DWARF entries to cover executable regions without DWARF info (e.g., PLT, hand-written assembly)
- Handle vdso and uprobes special memory regions:
  - Add support for [vdso] by reading ELF data directly from process memory via /proc/<pid>/mem
  - Add shard range validation in BPF program: stop unwinding when IP is not within any valid shard's [offset+pc_min, offset+pc_max) range
- Improve Python and Native stack merging strategy:
  - Adapt to Python 3.10+ vectorcall optimization: correctly handle cases where Python frame count exceeds _PyEval_EvalFrameDefault call count
  - Use Python interpreter stack as the authoritative call chain, only append C extension frames after the last _PyEval
  - Eliminate the previously frequent [lost] incomplete python c stack markers
- Fix Python symbol BPF Map iteration logic:
  - Correct bpf_get_next_key usage by using a separate next_key variable to prevent symbol loss
  - Add symbol count bounds checking to prevent array overflow
#### Affected branches
- main
#### Checklist
- [ ] Added unit test to verify the fix.
- [ ] Verified eBPF program runs successfully on linux 4.14.x.
- [ ] Verified eBPF program runs successfully on linux 4.19.x.
- [ ] Verified eBPF program runs successfully on linux 5.2.x.
- [x] Verified eBPF program runs successfully on linux 5.4.x.
- [x] Verified eBPF program runs successfully on linux 5.15.x.

### CODE REVIEW

**Comment by kylewanginchina** [comment]
原先存在的问题：
1、栈数据中的原生栈部分缺失libcuda.so、libcudart.so、libtorch_python.so等库函数相关的栈帧，显示[lost] incomplete python c stack；
2、栈数据中的python解释器栈部分得到的是错乱的python函数符号帧，与实际不符，并且对于同一进程同样负载，每次重启deepflow-agent后profile得到的栈帧还会变化；
3、缺少[[vdso]]相关库加载，对应原生栈无法正常展开；
4、deepflow-agent中进行python profile时会通过uprobe来hook python进程的处理，这个会反过来增加了目标python进程的采样栈帧，而[[uprobes]]对应的原生栈是不属于任何shard的，无法处理也不应该处理。
image
image

**Comment by kylewanginchina** [comment]
修复后的效果：
image
与otel一致：
image

采样比较少的部分的细节效果：
image
与otel一致：
image

**Comment by kylewanginchina** [comment]
uprobes额外说明：
image

**Comment by kylewanginchina** [comment]
多节点分布式推理这边看起来还有缺失python栈帧的问题，有待进一步处理解决：
img_v3_02st_c4f75f0e-8f7a-4cd1-9f87-29263690f5fg
img_v3_02st_88f63cae-32c8-4f2c-96cc-92c13da59d7g
应该和python多线程问题相关，因为从下面进程的这个火焰图大概可以看出来：
img_v3_02st_82c72bea-7223-4d9c-97fa-bdabd2b3b77g
